tests/tensorflow/test_tensorflow_estimator_export.py

```diff
@@ -0,0 +1,273 @@
+# pep8: disable=E501
+
+import collections
+import os
+import pickle
+import pytest
+import json
+
+import numpy as np
+import pandas as pd
+import pandas.testing
+import tensorflow as tf
+import iris_data_utils
+
+import mlflow
+import mlflow.pyfunc.scoring_server as pyfunc_scoring_server
+from mlflow import pyfunc
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.tensorflow import _TF2Wrapper
+from mlflow.utils.conda import get_or_create_conda_env
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.pyfunc.backend import _execute_in_conda_env
+from mlflow.tracking._tracking_service.utils import _use_tracking_uri
+
+from tests.helper_functions import pyfunc_serve_and_score_model, chdir
+
+
+@pytest.fixture
+def tf_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
+    _mlflow_conda_env(conda_env, additional_pip_deps=["tensorflow", "pytest"])
+    return conda_env
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(str(tmpdir), "model")
+
+
+ModelDataInfo = collections.namedtuple(
+    "ModelDataInfo",
+    [
+        "inference_df",
+        "expected_results_df",
+        "raw_results",
+        "raw_df",
+        "run_id",
+    ],
+)
+
+
+def save_or_log_tf_model_by_mlflow128(tmpdir, model_type, task_type, model_path=None):
+    conda_env = get_or_create_conda_env("tests/tensorflow/mlflow-128-tf-23-env.yaml")
+    output_data_file_path = os.path.join(tmpdir, "output_data.pkl")
+    tracking_uri = "file:" + os.path.abspath(os.path.join(tmpdir, "mlruns"))
+    exec_py_path = os.path.abspath("tests/tensorflow/save_tf_estimator_model.py")
+    mlflow_repo_path = os.getcwd()
+
+    with chdir(tmpdir):
+        # Change the current working directory to a temporary directory,
+        # to prevent importing current repo mlflow module.
+        _execute_in_conda_env(
+            conda_env,
+            f"python {exec_py_path} {model_type} {task_type} "
+            f"{model_path if model_path else ''}",
+            install_mlflow=False,
+            command_env={"MLFLOW_TRACKING_URI": tracking_uri, "MLFLOW_REPO_PATH": mlflow_repo_path},
```

Comment:
I think we can pass `tracking_uri` and `mlflow_repo_path` as arguments.

Reply:
What about passing all arguments via environment variables ? Extracting the arguments from sys.args is not good for code reading.

Reply:
You can use `argparse`.